app/tasks.py

```
@@ -401,401 +401,405 @@ def sync_all_users():
-            if battery_system != 'sigenergy' and sync_mode == 'initial_forecast' and getattr(user, 'force_tariff_mode_toggle', False):
+            # Check for firmware updates and notify if changed
+            firmware_version = site_info.get('version')
+            if firmware_version:
                 try:
-                    baseline_mode = tesla_client.get_operation_mode(user.tesla_energy_site_id)
-                    if baseline_mode:
-                        _sync_coordinator.set_baseline_mode(user.id, baseline_mode)
-                        logger.info(f"📍 Captured baseline operation mode for {user.email}: {baseline_mode}")
+                    from app.push_notifications import check_and_notify_firmware_change
+                    check_and_notify_firmware_change(user, firmware_version)
                 except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)